### DISCUSSION TITLE
Employ .NET [Auto]Vectorization features

### DISCUSSION THREAD

**Discussion by Mrnikbobjeff** [dotnet/maui#22612] [Ideas]
### Description

I took a look at MaciOS [Platformcanvas.cs](https://github.com/dotnet/maui/blob/b16a118a8b4e4df4d571d74e5e210d5202e2537e/src/Graphics/src/Graphics/Platforms/MaciOS/PlatformCanvas.cs#L345C28-L345C34) and noticed that we do some in theory vectorizable operations here. With the advent of System.Numerics.TensorPrimitives/VectorXXX<T> we should get some free speedups for little cost. I assume there are other instances where this holds, so this servers as an entry point to optimize different instances of vectorizable code.

### Public API Changes

None

### Intended Use-Case

Improved code generation employing intrinsics depending on ISA

**Comment by Mrnikbobjeff** [comment]
https://github.com/dotnet/maui/blob/b16a118a8b4e4df4d571d74e5e210d5202e2537e/src/Compatibility/Core/src/WPF/Microsoft.Windows.Shell/Standard/Utilities.cs#L32
exhibits several of these issues with their manual memcompare

**Reply by Xyncgas** [reply]
who's using a _MemCmp function in Microsoft.Windows.Shell.Standard.Utilities.cs I wonder
